### DISCUSSION TITLE
Container Fails to start

### DISCUSSION THREAD

**Discussion by Marauder0408** [haugene/docker-transmission-openvpn#1988] [Q&A] [answered]
Hello all,

I had this container running fine and then all of a sudden it would go unhealthy and a restart would fix it. Now it doesn’t work at all. I have tried googling and review the open issue and discussions but can’t figure it out. I use Portainer to create a stack and deploy the container, although I have also tried just running it in Docker and no difference. I am not very familiar with Docker or Linux in general so I am struggling here. Would appreciate any advice!

Config 

> version: '2'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
            - /home/user/mnt/mybooklive/Downloads:/data
        environment:
            - OPENVPN_PROVIDER=PIA
            - OPENVPN_CONFIG=ca_toronto
            - OPENVPN_USERNAME=Xxxxxxxx
            - OPENVPN_PASSWORD=Xxxxxxxx
            - LOCAL_NETWORK=192.168.1.0/24
            - PUID=0 #Change to your PUID
            - PGID=0 #Change to your PGID
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '9091:9091'
        image: haugene/transmission-openvpn `

Logs 

> Starting container with revision: 6ee67f8210d7f9ca17677c2b108390acc91ec5ab
Creating TUN device /dev/net/tun
Using OpenVPN provider: PIA
Running with VPN_CONFIG_SOURCE auto
Provider PIA has a bundled setup script. Defaulting to internal config
Executing setup script for PIA
Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.3QP7uQQ4NE
Extract OpenVPN config bundle into PIA directory /etc/openvpn/pia
Starting OpenVPN using config ca_toronto.ovpn
Modifying /etc/openvpn/pia/ca_toronto.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Setting OpenVPN credentials...
adding route to local network 192.168.1.0/24 via 172.20.0.1 dev eth0
Sun Oct  3 00:35:09 2021 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 19 2021
Sun Oct  3 00:35:09 2021 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Sun Oct  3 00:35:09 2021 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Sun Oct  3 00:35:09 2021 TCP/UDP: Preserving recently used remote address: [AF_INET]154.3.40.131:1198
Sun Oct  3 00:35:09 2021 UDP link local: (not bound)
Sun Oct  3 00:35:09 2021 UDP link remote: [AF_INET]154.3.40.131:1198
Sun Oct  3 00:35:09 2021 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
Sun Oct  3 00:35:09 2021 [toronto418] Peer Connection Initiated with [AF_INET]154.3.40.131:1198
Sun Oct  3 00:35:10 2021 OpenVPN ROUTE6: OpenVPN needs a gateway parameter for a --route-ipv6 option and no default was specified by either --route-ipv6-gateway or --ifconfig-ipv6 options
Sun Oct  3 00:35:10 2021 OpenVPN ROUTE: failed to parse/resolve route for host/network: 2000::/3
Sun Oct  3 00:35:10 2021 TUN/TAP device tun0 opened
Sun Oct  3 00:35:10 2021 /sbin/ip link set dev tun0 up mtu 1500
Sun Oct  3 00:35:10 2021 /sbin/ip addr add dev tun0 10.7.112.112/24 broadcast 10.7.112.255
Sun Oct  3 00:35:10 2021 /etc/openvpn/tunnelUp.sh tun0 1500 1553 10.7.112.112 255.255.255.0 init
Up script executed with tun0 1500 1553 10.7.112.112 255.255.255.0 init
Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.7.112.112
Updating Transmission settings.json with values from env variables
Using existing settings.json for Transmission /data/transmission-home/settings.json
Traceback (most recent call last):
  File "/etc/transmission/updateSettings.py", line 45, in <module>
    settings_dict = json.load(input_file)
  File "/lib/python3.8/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
Sun Oct  3 00:35:10 2021 WARNING: OpenVPN was configured to add an IPv6 route over tun0. However, no IPv6 has been configured for this interface, therefore the route installation may fail or may not work as expected.
Sun Oct  3 00:35:10 2021 Initialization Sequence Completed `

**Comment by pkishino** [comment, marked as answer]
if you check the log you will se the error..
there is a problem with your /data/transmission-home/settings.json file
check that the values inside are correct.  Possibly the first line is not correct.
please check, fix and then reload

**Reply by Marauder0408** [reply]
Thanks for the response. I’ve never touched the settings.json file, I will look into that.

It looks to me like some sort of error with ipv6, I tried again and still get these errors 

> 021-10-04 01:53:32 WARNING: OpenVPN was configured to add an IPv6 route. However, no IPv6 has been configured for tun0, therefore the route installation may fail or may not work as expected.
2021-10-04 01:53:32 add_route_ipv6(2000::/3 -> :: metric -1) dev tun0
RTNETLINK answers: Permission denied
2021-10-04 01:53:32 ERROR: Linux route -6 add command failed: external program exited with error status: 2
2021-10-04 01:53:32 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
2021-10-04 01:53:32 Initialization Sequence Completed

I had added the following based on some google suggestions but doesn’t seem to help

> sysctls:
            - net.ipv6.conf.all.disable_ipv6=1

**Reply by pkishino** [reply]
did adding the disable ipv6 at least get rid of the error ? 
as I mentioned, to me it looks like the transmissions settings.json isn't correctly formatted for something. can you please show the whole file please? (remove rpc password if you want..)

**Reply by Marauder0408** [reply]
No it didn’t change anything. My settings.json file appears to be empty. I have deleted it and tried creating the container, it just creates an empty file. I’ve tried viewing it in Notepad++ with the JSON viewer and tried in the command line with vi but still seems empty.

**Reply by pkishino** [reply]
hmm, that is very strange.. can you please go to the console in portainer and check "cat /data/transmission-home/settings.json" please? if it cannot create the settings..hmmm, permissions issue

**Reply by pkishino** [reply]
in portainer, your volumes look similar to this, rght?
https://github.com/haugene/docker-transmission-openvpn/discussions/1968#discussioncomment-1422210

**Reply by Marauder0408** [reply]
When I do that in the console, nothing happens. And yes my volumes look similar. Attached some pictures.

Everything worked no problem, then I started having issues where the container would go unhealthy and a simple restart would fix it for a bit. Now it doesn’t run at all. I did upgrade to the latest Linux a couple months ago but think it was still working after that.
![595A2AF5-A015-451F-9F02-1F55CF0C3E95](https://user-images.githubusercontent.com/17320936/136136466-0c52a1f4-a8e6-4150-8c67-260f1a6d5619.jpeg)
![DBA522DA-EFBC-4397-8E27-85276FE8F54F](https://user-images.githubusercontent.com/17320936/136136468-0113c04c-cc9b-467d-9f8d-a0e91dbb6cda.jpeg)

**Reply by pkishino** [reply]
please stop the container, delete the settings.json file and then start the container again, then check..
what are the permissions for the folder ?

**Reply by pkishino** [reply]
did you figure out the problem?

**Reply by Marauder0408** [reply]
Appreciate you following up, I haven't had time to check but went through it now. Stopped the container, deleted settings.json and restarted the container, still same error logs.

It is saving it onto my NAS as I have mounted it to my server. The permission of the folder is 'drwxr-xr-x 2 root root 'and the settings.json file is '-rwxr-xr-x 1 root root'.

Should I try having the transmission config saved directly to the server and not to the NAS? To be honest I'm not sure how to do that when setting up the volumes, the only volume I have is /home/ryan/mnt/mybooklive/Downloads:/data. Is there a way to assign two volumes, one for the container config files and one for the downloads?

**Reply by pkishino** [reply]
yes, you can have multiple mounts point each to explicit directories etc.. 
I suspect that you are having some kind of permissions clash here possibly..
have you tried removing PUID/PGID here to check, this will make it run as root..
otherwise you could have separate mounts kinda likes this:
volumes:
 - /home/user/mnt/mybooklive/Downloads:/data
 - /home/user/xxx/xxx:/data/transmission-home

or even something like this (as we want to migrate transmission-home default to /config
volumes:
 - /home/user/mnt/mybooklive/Downloads:/data
 - /home/user/xxx/xxx:/config

and then set
environment:
- TRANSMISSION_HOME=/config/transmission-home

**Reply by Marauder0408** [reply]
That worked! First tried just by removing PUID/PGID but did not change, then added the :/config and set the environment, works!

Now just have to get radarr/sonarr working again..

Appreciate your patience and walking me through it!

**Reply by pkishino** [reply]
hmm, interesting that that fixed it
good luck
